### PR TITLE
F/Add model_package_name for aws_sagemaker_model

### PR DESCRIPTION
**Pull request by holly-evans**
### Description
SageMaker `Model` object has support for passing a `ModelPackageName` to create models from model packages but this functionality was not implemented in the provider.

This PR adds this missing functionality.

Additionally, `image` is an optional argument for the `Containers` and `PrimaryContainer` structures.

### Relations

Closes https://github.com/hashicorp/terraform-provider-aws/issues/23968.
Closes https://github.com/hashicorp/terraform-provider-aws/issues/31533.
Closes https://github.com/hashicorp/terraform-provider-aws/pull/31532.
Closes https://github.com/hashicorp/terraform-provider-aws/pull/24004.

^ This PR combines these PRs and addresses the contributor feedback. @bschaatsbergen 

### References

https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateModel.html
https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ContainerDefinition.html

The AWS provider does not have a `aws_sagemaker_model_package` resource to create a new model package, so I used a free public model package in the test. I used a list from here - the models are region-specific.
https://github.com/aws/amazon-sagemaker-examples/blob/main/aws_marketplace/using_model_packages/creative-writing-using-gpt-2-text-generation/src/model_package_arns.py

### Output from Acceptance Testing

I am unable to run the tests on my AWS account.

### CODE REVIEW

File: .changelog/31755.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
+resource/aws_sagemaker_model: Add model_package_name argument
```
Comment (bschaatsbergen) [inline .changelog/31755.txt:2]:
```suggestion
resource/aws_sagemaker_model: Add `model_package_name` argument
```

File: internal/service/sagemaker/model_test.go [hunk 2]
```diff
@@ -618,6 +645,47 @@ resource "aws_s3_object" "test" {
 `, rName))
 }
 
+func testAccModelConfig_primaryContainerPackageName(rName string) string {
+	return acctest.ConfigCompose(testAccModelConfigBase(rName), fmt.Sprintf(`
+data "aws_region" "current" {}
```
Comment (bschaatsbergen) [inline internal/service/sagemaker/model_test.go:650]: Can you address the providerlint action comments here please?

**Review by bschaatsbergen — CHANGES_REQUESTED** [review]
Thanks for putting in the work so far @holly-evans, if you could kindly update the test we are close to getting this done :)

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccSageMakerModel_' PKG=sagemaker ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/sagemaker/... -v -count 1 -parallel 2  -run=TestAccSageMakerModel_ -timeout 180m
=== RUN   TestAccSageMakerModel_basic
=== PAUSE TestAccSageMakerModel_basic
=== RUN   TestAccSageMakerModel_inferenceExecution
=== PAUSE TestAccSageMakerModel_inferenceExecution
=== RUN   TestAccSageMakerModel_tags
=== PAUSE TestAccSageMakerModel_tags
=== RUN   TestAccSageMakerModel_primaryContainerModelDataURL
=== PAUSE TestAccSageMakerModel_primaryContainerModelDataURL
=== RUN   TestAccSageMakerModel_primaryContainerHostname
=== PAUSE TestAccSageMakerModel_primaryContainerHostname
=== RUN   TestAccSageMakerModel_primaryContainerImage
=== PAUSE TestAccSageMakerModel_primaryContainerImage
=== RUN   TestAccSageMakerModel_primaryContainerEnvironment
=== PAUSE TestAccSageMakerModel_primaryContainerEnvironment
=== RUN   TestAccSageMakerModel_primaryContainerModeSingle
=== PAUSE TestAccSageMakerModel_primaryContainerModeSingle
=== RUN   TestAccSageMakerModel_primaryContainerModelPackageName
=== PAUSE TestAccSageMakerModel_primaryContainerModelPackageName
=== RUN   TestAccSageMakerModel_containers
=== PAUSE TestAccSageMakerModel_containers
=== RUN   TestAccSageMakerModel_vpc
=== PAUSE TestAccSageMakerModel_vpc
=== RUN   TestAccSageMakerModel_primaryContainerPrivateDockerRegistry
=== PAUSE TestAccSageMakerModel_primaryContainerPrivateDockerRegistry
=== RUN   TestAccSageMakerModel_networkIsolation
=== PAUSE TestAccSageMakerModel_networkIsolation
=== RUN   TestAccSageMakerModel_disappears
=== PAUSE TestAccSageMakerModel_disappears
=== CONT  TestAccSageMakerModel_basic
=== CONT  TestAccSageMakerModel_primaryContainerModeSingle
--- PASS: TestAccSageMakerModel_primaryContainerModeSingle (36.49s)
=== CONT  TestAccSageMakerModel_primaryContainerHostname
--- PASS: TestAccSageMakerModel_basic (37.86s)
=== CONT  TestAccSageMakerModel_primaryContainerEnvironment
--- PASS: TestAccSageMakerModel_primaryContainerHostname (38.34s)
=== CONT  TestAccSageMakerModel_primaryContainerImage
--- PASS: TestAccSageMakerModel_primaryContainerEnvironment (44.24s)
=== CONT  TestAccSageMakerModel_tags
--- PASS: TestAccSageMakerModel_primaryContainerImage (45.37s)
=== CONT  TestAccSageMakerModel_primaryContainerModelDataURL
--- PASS: TestAccSageMakerModel_tags (70.62s)
=== CONT  TestAccSageMakerModel_primaryContainerPrivateDockerRegistry
--- PASS: TestAccSageMakerModel_primaryContainerModelDataURL (40.63s)
=== CONT  TestAccSageMakerModel_disappears
--- PASS: TestAccSageMakerModel_disappears (30.61s)
=== CONT  TestAccSageMakerModel_networkIsolation
--- PASS: TestAccSageMakerModel_primaryContainerPrivateDockerRegistry (41.08s)
=== CONT  TestAccSageMakerModel_containers
--- PASS: TestAccSageMakerModel_networkIsolation (34.56s)
=== CONT  TestAccSageMakerModel_vpc
--- PASS: TestAccSageMakerModel_containers (35.87s)
=== CONT  TestAccSageMakerModel_primaryContainerModelPackageName
--- PASS: TestAccSageMakerModel_primaryContainerModelPackageName (33.49s)
=== CONT  TestAccSageMakerModel_inferenceExecution
--- PASS: TestAccSageMakerModel_vpc (40.42s)
--- PASS: TestAccSageMakerModel_inferenceExecution (35.45s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/sagemaker	273.878s
```

**Comment by ewbankkit** [comment]
@holly-evans Thanks for the contribution :tada: :clap:.

**Comment by bschaatsbergen** [comment]
@ewbankkit 🥇

**Pull request merged**